static/src/javascripts/lib/cmp.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import {
+    getAdConsentState,
+    thirdPartyTrackingAdConsent,
+} from 'common/modules/commercial/ad-prefs.lib';
+
+type PurposeEvent = 'functional' | 'performance' | 'advertisement';
+
+type PurposeCallback = (state: boolean | null) => void;
+
+type Purpose = {
+    state: boolean | null,
+    callbacks: Array<PurposeCallback>,
+};
+
+let cmpIsReady = false;
+
+const purposes: { [PurposeEvent]: Purpose } = {
+    functional: {
+        state: null,
+        callbacks: [],
+    },
+    performance: {
+        state: null,
+        callbacks: [],
+    },
+    advertisement: {
+        state: null,
+        callbacks: [],
+    },
+};
+
+const checkCmpReady = (): void => {
+    if (cmpIsReady) {
+        return;
+    }
+
+    /**
+     * These state assignments are temporary
+     * and will eventually be replaced by values
+     * read from the CMP cookie.
+     * */
+    purposes.functional.state = true;
+    purposes.performance.state = true;
+    purposes.advertisement.state = getAdConsentState(
+        thirdPartyTrackingAdConsent
+    );
+
+    cmpIsReady = true;
+};
+
+const triggerConsentNotification = (): void => {
+    checkCmpReady();
+
+    Object.keys(purposes).forEach(key => {
```

Comment:
Is that OK if purposes are triggered in random orders?

Reply:
We made a decision to keep this library naive to the execution order, it's the modules that pass the callbacks that should account for that.

Reply:
Yes but here, each purpose will be triggered in an undefined order. I suppose that's ok :)

Reply:
That's right. There's no dependencies between modules in different purposes so the order we trigger these is irrelevant.